test/falcor/get/get.dataSource-only.spec.js

```diff
@@ -275,6 +275,89 @@ describe('DataSource Only', function() {
         }, 200);
     });
 
+    it('should ignore response-stuffed paths.', function(done) {
+        var onGet = sinon.spy();
+        var source = new LocalDataSource(cacheGenerator(0, 2), {
+            onGet: onGet,
+            wait: 100
+        });
+        var model = new Model({source: source}).batch(1);
+        var onNext = sinon.spy();
+        var disposable1 = toObservable(model.
+            get(['videos', 0, 'title'])).
+            doAction(onNext, noOp, function() {
+                throw new Error('Should not of completed.  It was disposed.');
+            }).
+            subscribe(noOp, done);
+
+        toObservable(model.
+            get(['videos', 1, 'title'])).
+            subscribe(noOp, done);
+
+        setTimeout(function() {
+            disposable1.dispose();
+        }, 30);
+
+        setTimeout(function() {
+            try {
+                expect(model._root.cache.videos[0]).to.be.undefined;
+            } catch(e) {
+                return done(e);
+            }
+            return done();
+        }, 200);
+    });
+
+    it('should honor response-stuffed paths with _useServerPaths == true.', function(done) {
+        var onGet = sinon.spy();
+        var source = new LocalDataSource(cacheGenerator(0, 2), {
+            onGet: onGet,
+            wait: 100,
+            onResults: function(data) {
+                data.paths = [
+                    ['videos', 0, 'title'],
+                    ['videos', 1, 'title']
+                ];
```

Comment:
I imagine this might be collapsed server side to `[['videos', { from: 0, to: 1 }, 'title']]`.

Reply:
Not a problem, as tons of tests already cover the case of complex keys (and they remain complex keys in _merge).